woocommerce-services.php

```diff
@@ -1898,22 +1900,52 @@ public function display_woo_shipping_and_woo_tax_are_active_notice() {
 		public function display_wcst_to_wcshipping_migration_notice() {
 			$schema = $this->get_service_schemas_store();
 			$banner = $schema->get_wcship_wctax_upgrade_banner();
+
+			$account_settings  = new WC_Connect_Account_Settings(
+				$this->service_settings_store,
+				$this->payment_methods_store
+			);
+			$packages_settings = new WC_Connect_Package_Settings(
+				$this->service_settings_store,
+				$this->service_schemas_store
+			);
+			$encoded_arguments = wp_json_encode(
+				array(
+					'nonce'            => wp_create_nonce( 'wp_rest' ),
+					'baseURL'          => get_rest_url(),
+					'accountSettings'  => $account_settings->get(),
+					'packagesSettings' => $packages_settings->get(),
+				)
+			);
+
 			echo wp_kses_post(
 				sprintf(
-					'<div class="notice notice-%s is-dismissible wcst-wcshipping-migration-notice"><p style="margin-bottom:0px">',
-					$banner->type
+					'<div class="notice notice-%s is-dismissible wcst-wcshipping-migration-notice">
+					<div id="wcst_wcshipping_migration_admin_notice_feature_announcement" data-args="%s"></div>
+					<p class="wcst-wcshipping-migration-notice__p">',
+					$banner->type,
+					wc_esc_json( $encoded_arguments )
 				) .
 				sprintf(
 					/* translators: %s: documentation URL */
 					__( $banner->message, 'woocommerce-services' ),
 					'https://woocommerce.com/document/woocommerce-shipping-and-tax/woocommerce-shipping/#how-do-i-migrate-from-wcst'
 				) .
 				sprintf(
-					'</p><button style="color:#3858E9;margin: 12px 0;border: 1px solid #3858E9;padding: 11.5px 12px 11.5px 12px;border-radius: 2px;background-color:#fff;">Confirm update</button></div>',
+					'</p><button id="wcst-wcshipping-migration-notice__click" class="button action wcst-wcshipping-migration-notice__button">Confirm update</button></div>',
 					$banner->action
 				)
 			);
 		}
+
+		public function register_wcshipping_migration_modal() {
+			$plugin_version = self::get_wcs_version();
+			wp_register_style( 'wcst_wcshipping_migration_admin_notice', $this->wc_connect_base_url . 'woocommerce-services-wcshipping-migration-admin-notice-' . $plugin_version . '.css', array(), null ); // phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion
+			wp_register_script( 'wcst_wcshipping_migration_admin_notice', $this->wc_connect_base_url . 'woocommerce-services-wcshipping-migration-admin-notice-' . $plugin_version . '.js', array(), null );
+			wp_enqueue_script( 'wcst_wcshipping_migration_admin_notice' );
+			wp_enqueue_style( 'wcst_wcshipping_migration_admin_notice' );
+			wp_enqueue_script( 'wc_connect_admin' );
```

Comment:
> wp_enqueue_script( 'wc_connect_admin' );

I enqueued this whole script because `<FeatureAnnouncement>` needs the global `global.wcsPluginData.adminPluginPath`, which is a variable in the `wc_connect_admin` script. Loading this whole script is easier.

Alternatively, I could have extracted the `global.wcsPluginData.adminPluginPath`, and loads it in the [$encoded_arguments](https://github.com/Automattic/woocommerce-services/pull/2756/files#diff-4cb5834c68b324f4fbecc437b25f62021e55ecc28db9adb93053eca0391f5cc5R1912) up there. If I do this, I would have to edit `<FeatureAnnouncement>` and change:
 `window.location = global.wcsPluginData.adminPluginPath;`
to something like:
`window.location = global.wcsPluginData.adminPluginPath || someState.adminPluginPath;`.
This doesn't seem worth it because the condition is more complicated. So I chose to load the whole `wc_connect_admin` script into the order list page instead.

Reply:
Thanks for the explanation. This also seems OK performance-wise because I think `/client/main.js` won't load the specific functionality (like the label purchase modal) unless certain conditions are met, correct? 🤔

Reply:
Thanks @waclawjacek! Actually, you just made me realize that `client/main.js` is the registered `wc_connect_admin` script which turns into `'woocommerce-services-*.js`. 
If I am already loading `wc_connect_admin`, maybe we didn't need its own entry point. I have 3 options.

#### Option 1.
Leave it as is because this is the migration modal only. Loading `'woocommerce-services-*.js` for `adminPluginPath` is heavy but it is also cached, so it should be faster for the individual order page. The `client/wcshipping-migration-admin-notice.js` is separated as if it was chunk-split and won't be loaded inside the individual order page, which is good.

#### Option 2.
Remove `client/wcshipping-migration-admin-notice.js` and move that into `main.js`. Use `enqueue_wc_connect_script()` pattern to create the `wcc-` container inside the admin notice in order to load this `<FeatureAnnouncement>` modal.

#### Option 3.
Don't load `wp_enqueue_script( 'wc_connect_admin' );`. Pull `adminPluginPath` and find a way to load this in the order list page, and then inject it to the `<FeatureAnnouncement>` modal.

I am opting for **Option 1** because there is a slight performance hit on the order list page, but does not impact overall total performance since `woocommerce-service-*.js` is cached and will be used in the individual order page, it will be loaded regardless. On size, it will actually be better because these codes won't be loaded on the individual order page.

Let me know your thoughts!

Reply:
I'd rather go with **Option 1** as it's a migration specific change.